ee/src/main/java/org/jboss/as/ee/component/deployers/EEModuleConfigurationProcessor.java

```diff
@@ -106,6 +107,16 @@ public void deploy(DeploymentPhaseContext phaseContext) throws DeploymentUnitPro
         }
     }
 
+    private static boolean isTopLevel(DeploymentUnit deploymentUnit) {
+        return deploymentUnit.getParent() == null;
+    }
+
+    private static DeploymentUnit getTopLevel(DeploymentUnit deploymentUnit) {
+        DeploymentUnit result = deploymentUnit;
+        while (!isTopLevel(result)) result = result.getParent();
```

Comment:
There is no need for a loop, there can only ever be two levels. Something like:

top = du.getParent() != null ? du.getParent() : du;

Is the standard way of doing this.